sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -1096,3 +1097,54 @@ async def test_queue_message_settle_through_mgmt_link_due_to_broken_receiver_lin
                 await receiver._handler.message_handler.destroy_async()  # destroy the underlying receiver link
                 assert len(messages) == 1
                 await messages[0].complete()
+
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    @ServiceBusQueuePreparer(name_prefix='servicebustest')
+    async def test_async_queue_by_servicebus_conn_str_autocomplete(self, servicebus_namespace_connection_string, servicebus_queue, **kwargs):
+
+        async with ServiceBusClient.from_connection_string(
+            servicebus_namespace_connection_string, logging_enable=False) as sb_client:
+
+            async with sb_client.get_queue_sender(servicebus_queue.name) as sender:
+                await sender.send(Message("Test Message"))
+                await sender.send(Message("Test Message"))
+                #Intentionally sending twice to test both auto and manual complete interaction.
+
+            count = 0
+            async with sb_client.get_queue_receiver(servicebus_queue.name, idle_timeout=1) as receiver:
+                for message in await receiver.receive(1):
+                    async with AutoComplete(message):
+                        print_message(_logger, message)
+                        count += 1
+                        await message.complete() # It should be fine with if we complete manually as well.
+            assert count == 1
+
+            count = 0
+            async with sb_client.get_queue_receiver(servicebus_queue.name, idle_timeout=5) as receiver:
+                try:
+                    async for message in receiver:
+                        async with AutoComplete(message):
+                            print_message(_logger, message)
+                            count += 1
+                            raise AssertionError("Intentional Failure to trigger auto-abandon.")
+                except AssertionError as e:
+                    print(e)
+                    pass
+            assert count == 1
```

Comment:
nitpick: should "try-except" be in the for-loop?

```python
async for message in receiver:
    try:
        async with AutoComplete(message):
            print_message(_logger, message)
            count += 1
            raise AssertionError("Intentional Failure to trigger auto-abandon.")
    except AssertionError as e:
        print(e)
        pass
assert count == 1
```

Reply:
Nope, that was intentional. (although it wouldn't make a difference if put inside/outside, if I'm understanding properly)  If you look above, I send 2 messages, complete one manually to check double-settlement, let this one abandon to test auto-abandon, then re-receive and auto-complete this final one.